api/libretime_api/storage/views/file.py

```diff
@@ -29,3 +29,18 @@ def download(self, request, pk=None):  # pylint: disable=invalid-name
         redirect_uri = filepath_to_uri(os.path.join("/api/_media", file.filepath))
         response["X-Accel-Redirect"] = redirect_uri
         return response
+
+    @action(detail=True, methods=["DELETE"])
+    def delete_file(self, request, pk=None):  # pylint: disable=invalid-name
+        pk = IntegerField().to_internal_value(data=pk)
+
+        file = get_object_or_404(File, pk=pk)
+        path = file.filepath
+
+        try:
+            if os.path.isfile(path):
+                os.remove(path)
```

Comment:
> Right now the code will only manipulate the database but leave the file in place.

This line implies the opposite? It will remove the actual file

Reply:
I worded that poorly. What i wanted to say is, that the file deletion is introduced with this patch, and without the patch the database is updated but the file stays on storage. 

Use case is an external script that prunes old podcast episodes from the server.

Reply:
@caveman99  libretime/api/schema.yml may require change to pass check

Reply:
A `make -C api schema` from the root of the project should fix it.